specs/node_types.md

```diff
@@ -0,0 +1,105 @@
+
+### Block Bodies
+
+Block bodies (the `availableData` [block](./data_structures.md#block) field) can be downloaded and optionally stored and/or served. Storing and serving block body data has no effect on node security assumptions.
+
+#### No Bodies
+
+Nodes that only process [compact block headers](#compact-block-headers) have no need for block bodies and simply do not process block bodies.
+
+Secure under an honest supermajority majority of validator voting power and a weak subjectivity assumption.
+
+#### Sampled Bodies
+
+These nodes perform Data Availability Sampling on block bodies.
+
+Secure under an honest minority of nodes and a weak subjectivity assumption.
+
+#### Partial Bodies
+
+These nodes fully download and validate [the erasure coding](./data_structures.md#2d-reed-solomon-encoding-scheme) of a random subset of block bodies (configurable locally). Since the erasure coding of each block is stateless, nodes that perform validation of partial bodies contribute to the overall security of the network by being able to produce [fraud proofs of invalid erasure coding](./data_structures.md#invalid-erasure-coding).
+
+Secure under an honest minority of nodes and a weak subjectivity assumption.
+
+#### Full Bodies
+
+These nodes fully download and validate [the erasure coding](./data_structures.md#2d-reed-solomon-encoding-scheme) of all block bodies.
+
+If [transactions are not processed](#no-transactions), secure under an honest minority of nodes and a weak subjectivity assumption. If [transactions are processed](#full-transactions), secure under a weak subjectivity assumption.
+
+### Transactions
+
+#### No Transactions
+
+These nodes process do not process requests [with a reserved namespace ID](./data_structures.md#arranging-available-data-into-shares) and thus to not know the chain state without relying on a third party.
+
+At most secure under an honest minority of nodes and a weak subjectivity assumption.
+
+#### Full Transactions
+
+Nodes that wish to produce new blocks must know the [chain state](./data_structures.md#state). Processing all block bodies is actually not needed to know the LazyLedger state, as [transactions that pay for message inclusion commit to messages](./../rationale/message_block_layout.md). These nodes process all requests [with a reserved namespace ID](./data_structures.md#arranging-available-data-into-shares) from block bodies and perform Data Availability Sampling for the remaining (message) data.
+
+At most secure under a weak subjectivity assumption.
+
+## Node Types
+
+For convenience, we will define several common parameter configurations:
+1. [Full nodes](https://en.bitcoin.it/wiki/Full_node) provide the strongest security guarantees. Block bodies are not stored.
```

Comment:
> Block bodies are not stored.

But they can be?

Reply:
Any user is free to configure these parameters differently, but the expression "full nodes" should be reserved for the node type that only keep track of the latest state and not the entire history. It's been a point of confusion with other blockchains that we should draw the line on.

Reply:
Nonetheless, clarified wording in e5d2320.